mazerunner-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/UnhandledExceptionApiKeyChangeScenario.kt

```diff
@@ -0,0 +1,26 @@
+package com.bugsnag.android.mazerunner.scenarios
+
+import android.content.Context
+
+import com.bugsnag.android.Configuration
+import com.bugsnag.android.OnErrorCallback
+
+/**
+ * Sends an unhandled exception to Bugsnag where the API key is changed in a callback
+ */
+internal class UnhandledExceptionApiKeyChangeScenario(config: Configuration,
+                                                      context: Context) : Scenario(config, context) {
+    init {
+        config.autoTrackSessions = false
+        config.addOnError(OnErrorCallback { event ->
+            event.apiKey = "0000111122223333aaaabbbbcccc9999"
```

Comment:
Suggestion: might be worth adding another scenario where one client instance generates an event with a specific API key (without overriding it in OnErrorCallback, just using its default config) and dumps it on the file system and then a new client instance is created with a different API key, and that new client instance ends up uploading the error and uses the right (initial) API key.